torch/csrc/autograd/saved_variable.cpp

```diff
@@ -171,6 +209,37 @@ void SavedVariable::register_hooks(std::unique_ptr<SavedVariableHooks>&& hooks)
     "Calling register_hooks on a saved tensor whose hooks have already been set. "
     "Hint: only one pair of hooks is allowed at a time.");
   hooks_ = std::move(hooks);
+
+  // If we didn't save the original variable, we already have all we need to reconstruct it
+  if (saved_original_) {
+    // TODO(varal7) refactor the next 10 lines
+    output_nr_ = data_.output_nr();
+    version_counter_ = impl::version_counter(data_);
+
+    // TODO(albanD) This needs to be updated when moving to multiple levels
+    const auto& fw_grad = data_._fw_grad(/* level */ 0);
+    if (fw_grad.defined()) {
+      fw_grad_ = std::make_shared<ForwardGrad>();
+      fw_grad_->set_value(fw_grad, /* level */ 0);
+    }
+
+    if (is_leaf_) {
+      grad_accumulator_ = impl::grad_accumulator(data_);
+      requires_grad_ = data_.requires_grad();
+    } else if (!is_output_) {
+      grad_fn_ = data_.grad_fn();
+    } else {
+      // Current code assumes that the original variable is saved if and only if (is_leaf_ || !is_output)
+      TORCH_INTERNAL_ASSERT(false);
+    }
+    data_ = data_.tensor_data();
+  }
+
+  const bool prev = GradMode::is_enabled();
```

Comment:
You can use `NoGradGuard guard;` to disable autograd as long as that variable exists.